stingray/crosscorrelation.py

```diff
@@ -74,23 +82,65 @@ def __init__(self, lc1=None, lc2=None, mode='same'):
         self.mode = mode.lower()
         self.lc1 = None
         self.lc2 = None
+        self.cross = None
 
         # Populate all attributes by ``None` if user passes no lightcurve data
         if lc1 is None or lc2 is None:
             if lc1 is not None or lc2 is not None:
                 raise TypeError("You can't do a cross correlation with just one "
                                 "light curve!")
+
             else:
-                # both lc1 and lc2 are ``Non.
-                self.corr = None
-                self.time_shift = None
-                self.time_lags = None
-                self.dt = None
-                self.n = None
-                return
+                if cross is None:
+                    # all object input params are ``None``
+                    self.corr = None
+                    self.time_shift = None
+                    self.time_lags = None
+                    self.dt = None
+                    self.n = None
+                else:
+                    self._make_cross_corr(cross)
+                    return
         else:
             self._make_corr(lc1, lc2)
 
+    def _make_cross_corr(self, cross):
+
+        """
+        Do some checks on the cross spectrum supplied to the method,
+        and then calculate the time shifts, time lags and cross correlation.
+
+        Parameters
+        ----------
+        cross: :class:`stingray.Crossspectrum` object
+            The crossspectrum, averaged or not.
+
+        """
+
+        if not isinstance(cross, Crossspectrum):
+            if not isinstance(cross, AveragedCrossspectrum):
+                raise TypeError("cross must be a crossspectrum.Crossspectrum \
+                        or crossspectrum.AveragedCrossspectrum object")
+
+        if self.cross is None:
```

Comment:
I'm wondering why this if-statement is here. Are there ever situations in this function where `self.cross` is not `None`?